gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/BigQuery.java

```diff
@@ -25,5 +34,602 @@
+     */
+    public static TableDataListOption maxResults(long maxResults) {
+      return new TableDataListOption(BigQueryRpc.Option.MAX_RESULTS, maxResults);
+    }
+
+    /**
+     * Returns an option to specify the page token from which to start listing table data.
+     */
+    public static TableDataListOption startPageToken(String pageToken) {
+      return new TableDataListOption(BigQueryRpc.Option.PAGE_TOKEN, pageToken);
+    }
+
+    /**
+     * Returns an option that sets the zero-based index of the row from which to start listing table
+     * data.
+     */
+    public static TableDataListOption startIndex(long index) {
+      return new TableDataListOption(BigQueryRpc.Option.START_INDEX, index);
+    }
+  }
+
+  /**
+   * Class for specifying job list options.
+   */
+  class JobListOption extends Option {
+
+    private static final long serialVersionUID = -8207122131226481423L;
+
+    private JobListOption(BigQueryRpc.Option option, Object value) {
+      super(option, value);
+    }
+
+    /**
+     * Returns an option to list all jobs, even the ones issued by other users.
+     */
+    public static JobListOption allUsers() {
+      return new JobListOption(BigQueryRpc.Option.ALL_USERS, true);
+    }
+
+    /**
+     * Returns an option to list only jobs that match the provided filters.
+     */
+    public static JobListOption stateFilter(JobStatus.State... stateFilters) {
+      List<String> stringFilters = Lists.transform(ImmutableList.copyOf(stateFilters),
+          new Function<JobStatus.State, String>() {
+            @Override
+            public String apply(JobStatus.State state) {
+              return state.toString().toLowerCase();
+            }
+          });
+      return new JobListOption(BigQueryRpc.Option.STATE_FILTER, stringFilters);
+    }
+
+    /**
+     * Returns an option to specify the maximum number of jobs to be returned.
+     */
+    public static JobListOption maxResults(long maxResults) {
+      return new JobListOption(BigQueryRpc.Option.MAX_RESULTS, maxResults);
+    }
+
+    /**
+     * Returns an option to specify the page token from which to start listing jobs.
+     */
+    public static JobListOption startPageToken(String pageToken) {
+      return new JobListOption(BigQueryRpc.Option.PAGE_TOKEN, pageToken);
+    }
+
+    /**
+     * Returns an option to specify the job's fields to be returned by the RPC call. If this option
+     * is not provided all job's fields are returned. {@code JobOption.fields()}) can be used to
+     * specify only the fields of interest. {@link JobInfo#jobId()}, {@link JobStatus#state()},
+     * {@link JobStatus#error()} as well as type-specific configuration (e.g.
+     * {@link QueryJobInfo#query()} for Query Jobs) are always returned, even if not specified.
+     * {@link JobField#SELF_LINK} and {@link JobField#ETAG} can not be selected when listing jobs.
+     */
+    public static JobListOption fields(JobField... fields) {
```

Comment:
Do you think this is useful (considering the possible return values and the minimum that we specify)? I am not sure about it, and not having it would be consistent with the other listing operations that do not provide a field selector.